src/Compilers/CSharp/Portable/CSharpCompilationOptions.cs

```diff
@@ -211,7 +226,9 @@ public sealed class CSharpCompilationOptions : CompilationOptions, IEquatable<CS
             bool referencesSupersedeLowerVersions,
             bool publicSign,
             BinderFlags topLevelBinderFlags,
-            NullableContextOptions nullableContextOptions)
+            NullableContextOptions nullableContextOptions,
+            Encoding? codePage,
+            IEnumerable<string>? preprocessorSymbols)
```

Comment:
I'm not sure this is the best approach. I'd rather not add compilation options that duplicate the information that's already in syntax trees, especially since they are only used during emit. 
Other approaches:
1) read the values from the first syntax tree
2) add these to EmitOptions

Re [1] - not sure if we can get the original encoding from the syntax tree. Need to look into that. But we can definitely get the preprocessor symbols from there.

Reply:
I'll remove preprocessor symbols, still unsure about encoding.

Reply:
OK, so I think there are two options for the encoding:
1) SourceText could remember the original encoding it was given (for the case when it switches to a different encoding based on BOM). Then we could read the codepage encoding from the first syntax tree given to the compilation.
2) Add the encoding to EmitOptions (btw, I'd call it `DefaultSourceFileEncoding` rather than codepage)

I'd go with [2] as it is simpler.